src/iam/iam-base.ts

```diff
@@ -673,4 +677,31 @@ export class IAMBase {
       })
     });
   }
+
+  private async initContracts() {
+    const { chainId } = await this._provider.getNetwork();
+    const {
+      ensRegistryAddress, ensResolverAddress, didContractAddress, cacheServerUrl
+    } = IAMBase.chainConfigs[chainId];
+    this._registrySetting = {
+      address: didContractAddress,
+      abi: abi1056,
+      method: Methods.Erc1056
+    };
+    this._ensRegistryAddress = ensRegistryAddress;
+    this._ensResolverAddress = ensResolverAddress;
+    this._ensRegistry = EnsRegistryFactory.connect(ensRegistryAddress, this._provider);
+    this._ensResolver = PublicResolverFactory.connect(ensResolverAddress, this._provider);
+    cacheServerUrl && (this._cacheClient = new CacheServerClient({ url: cacheServerUrl }));
```

Comment:
I think this is a bit of a problem, as `CacherServerClient` has another initialization property (`cacheServerSupportAuth`) and could have others in the future. Maybe it is better just to continue to pass in the `CacheServerClient` in the constructor for now?

Reply:
Hmm, I think I see what you're saying. You're saying that by constructing a default cache-server like this, if for example, the user changed their network in metamask, we could connect to the correct cache-server for them.

How about this design? 
- Have an interface of `CacheServerClientOptions` that is used for `CacheServerClient` initialization
- Have `ChainConfig` extend `CacheServerClientOptions`
- In this way, we can have a default for each chain, but also apps can override their cache-server config if they want

Reply:
Agree. Only suggestion - is to move cache client server options from chain config to allow partial update of both of them